standard/logic.lua

```diff
@@ -70,6 +70,22 @@ function Logic.isNotEmpty(val)
 	end
 end
 
+---Checks if a given object (table|string|nil) is deep empty
+---i.e. is empty itself or only contains objects that are deep empty
+---@param val table|string|number|nil
```

Comment:
Comment an param doesn't match

Could use the same param and overload as isEmpty.